cli/cmd/plugin/ui/web/tanzu-ui/node-server/src/routes/api/endpoints/vsphere.js

```diff
@@ -0,0 +1,189 @@
+        path: '/path/to/config'
+    });
+});
+
+/**
+ * Mock route for create vsphere cluster
+ */
+router.post('/create', (req, res) => {
+    winston.info('Mock UI CREATE VSPHERE CLUSTER');
+    res.status(200);
+    res.json({});
+});
+
+/**
+ * Retrieve compute resources
+ */
+router.get('/compute', (req, res) => {
+    winston.info('Mock UI RETRIEVE COMPUTE RESOURCES');
+    res.status(200);
+    res.json(readFile('provider-vsphere-compute.json'));
+});
+
+/**
+ * Mock route for connect VC server
+ */
+router.post('/', (req, res) => {
+    winston.info('Mock UI CONNECT VC API');
+    if ((req.body.host === 'vsphere.local' || req.body.host === '1.1.1.1' || req.body.host === '2001:0db8:85a3:0000:0000:8a2e:0370:7334') &&
+        (req.body.username === 'admin' || req.body.username === 'administrator') &&
+        (req.body.password === 'password')) {
+        res.status(200);
+        res.json({
+            'version': '6.7.0:14367737',
+            'hasPacific': 'no',
+            'tkrVersion': 'v1.17'
+        });
+    } else {
+        res.status(403);
+        res.json({ message: 'incorrect username or password' });
+    }
+});
+
+/**
+ * Mock route for getting VC datacenter network
+ */
+router.get('/networks', (req, res) => {
+    winston.info('Mock UI VC NETWORK API');
+
+    let vcNetworksResponse = readFile('provider-vsphere-networks.json');
+
+    if (mockvcNetworkRequestCounter > 0) {
+        vcNetworksResponse.push({
+            name: 'Network 7',
+            id: 'network-7',
+            displayName: 'Network 7'
+        });
+    }
+
+    mockvcNetworkRequestCounter++;
+
+    res.status(200);
+    res.json(vcNetworksResponse);
+});
+
+/**
+ * Mock route for thumbsprint
+ */
+
+router.get('/thumbprint', (req, res) => {
+    winston.info('Mock UI FETCH THUMBPRINT');
+    res.status(200);
+    res.json(readFile('provider-vsphere-thumbprint.json'));
+});
+
+/**
+ * Mock route for getting VC datacenters
+ */
+router.get('/datacenters', (req, res) => {
+    winston.info('Mock UI FETCH DATACENTERS');
+    res.status(200);
+    res.json(readFile('provider-vsphere-datacenters.json'));
+});
+
+/**
+ * Mock route for getting datastores
+ */
+router.get('/datastores', (req, res) => {
+    winston.info('Mock UI FETCH DATASTORES');
+    res.status(200);
+    res.json(readFile('provider-vsphere-datastores'));
+});
+
+/**
+ * Mock route for getting vm folders
+ */
+router.get('/folders', (req, res) => {
+    winston.info('Mock UI FETCH VM FOLDERS');
+    res.status(200);
+    if (req.body.dc === 'dc-1') {
+        res.json(readFile('provider-vsphere-folders-1.json'));
+    } else {
+        res.json(readFile('provider-vsphere-folders-2.json'));
+    }
+});
+
+/**
+ * Mock route for getting VC datacenters
+ */
+router.get('/resourcepools', (req, res) => {
+    winston.info('Mock UI FETCH RESOURCE POOLS');
+
+    let resourcePoolsResponse = readFile('provider-vsphere-resourcepools.json');
+
+    if (mockResourcePoolRequestCounter > 0) {
```

Comment:
I'd rather have all the knowledge of the data structure be in the JSON files.
So if we're going to have two different responses based on the number of times, I'd kinda like to see two different files, and then just choose the file here (rather than append a particular data structure). Curious what you and @miclettej think)

Reply:
good idea. I think two files are better.